modules/indexing/src/test/java/org/apache/ignite/internal/processors/cache/IgniteCachePutNotNullFieldTest.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.processors.cache;
+
+import java.io.Serializable;
+import java.util.Collections;
+import org.apache.ignite.IgniteCache;
+import org.apache.ignite.cache.CacheAtomicityMode;
+import org.apache.ignite.cache.QueryEntity;
+import org.apache.ignite.internal.processors.query.IgniteSQLException;
+import org.apache.ignite.testframework.GridTestUtils;
+import org.apache.ignite.testframework.junits.common.GridCommonAbstractTest;
+
+/**
+ *
+ */
+public class IgniteCachePutNotNullFieldTest extends GridCommonAbstractTest {
+    /**
+     * @throws Exception If failed.
+     */
+    @SuppressWarnings("ThrowableNotThrown")
+    public void testPutAllShouldThrowExceptionWhenPassedNullValue() throws Exception {
+        final IgniteCache<Integer, Address> cache = jcache(
+            startGrid(0),
+            defaultCacheConfiguration()
+                .setAtomicityMode(CacheAtomicityMode.ATOMIC)
+                .setQueryEntities(
+                    Collections.singleton(
+                        new QueryEntity(Integer.class.getName(), Address.class.getName())
+                            .addQueryField("address", "java.lang.String", "address")
+                            .setNotNullFields(Collections.singleton("address"))
+                    )
+                ),
+            "ORG_ADDRESS"
+        );
+
+        GridTestUtils.assertThrowsWithCause(
+            () -> {
+                cache.putAll(Collections.singletonMap(1, new Address(null)));
+
+                return null;
+            },
+            IgniteSQLException.class
+        );
+    }
+
+    /** */
+    private static class Address implements Serializable {
+        /** */
+        private final String addr;
```

Comment:
`.addQueryField("address", "java.lang.String", "address")`
and 
`private final String addr;`
from my point of view should have the same valuable name.